### PR TITLE
[GEOS-10788] CITE WCS 1.1.1 -  Empty InterpolationMethod should throw exception

### PR DESCRIPTION
**Pull request by davidblasby**
[GEOS-10788](https://osgeo-org.atlassian.net/browse/GEOS-10788)

While running the WCS 1.1.1 CITE tests, GS wasn't passing some of the interpolationMethod tests.

This was discussed with OGC CITE - https://github.com/opengeospatial/ets-wcs11/issues/76

Because the spec is not specific, the CITE test will be changed to only check for an exception report.

The PR only addresses the outstanding problem - ensuring that GS throws an exception where there is a blank interpolationMethod.

Here is an example CITE test;

https://github.com/opengeospatial/ets-wcs11/blob/master/src/main/scripts/ctl/GetCoverage.xml#L5586

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Review by jodygarnett — COMMENTED** [review]
David this looks good to me, but I am not comfortable approving a change  in the WCS module.

**Comment by aaime** [comment]
The changes look good in general, what is surprising me is the syntax of the locator. Normally the locator is the name of the key that would be involved in the error, if the request was made using a KVP GET request. This one looks more similar to an XML XPath instead, which I don't recall seeing before. @davidblasby what are the CITE tests actually looking for? Specifically for the form you used?

Also, according to the spec, the interpolation specification is not mandatory:

image

The test seems to be checking whether the interpolation is present but an empty string, which of course is not valid... but why not check for invalid values instead? The change seems more of a CITE tests workaround, but mind, often times the tests themselves are wrong (I've reported many to OGC while playing with CITE tests, over the years).

**Comment by davidblasby** [comment]
Andrea,

I was surprised at the syntax of the locator as well. 

Here an example of one of their tests;
https://github.com/opengeospatial/ets-wcs11/blob/master/src/main/scripts/ctl/GetCoverage.xml#L5586

That test case has the request look like `content:`, which is invalid.

That same test has a XML Post:
https://github.com/opengeospatial/ets-wcs11/blob/master/src/main/scripts/ctl/GetCoverage.xml#L5573

 `<InterpolationType/>` - which is "present but null"

There are a few other tests that are similar.

GS was, previously, ignoring some of these issues.  I also made the errors a bit more specific.  

The specification is ambiguous on what the actual error message should be.  I expect the CITE test is incorrect (the text looks too specific), but there isn't anything in the specification that says what it should be.

**Comment by davidblasby** [comment]
I put a bug report in the OGC github:
https://github.com/opengeospatial/ets-wcs11/issues/76

@aaime - perhaps comment there?

**Comment by davidblasby** [comment]
Looks like the OGC will be discussing this - lets put this PR to sleep until they give us guidance.

**Comment by davidblasby** [comment]
I am making an ogc-wcs CITE PR to change the test case behaviour.

However, there still needs to be changes in the GS code base because its not throwing exceptions when it should.

I'm going to add part of this pull request in and have it throw an exception if they have the missing info.

**Comment by davidblasby** [comment]
see https://github.com/opengeospatial/ets-wcs11/pull/77

**Comment by davidblasby** [comment]
DO NOT MERGE UNTIL OGC MERGES

**Comment by davidblasby** [comment]
OGC has merged https://github.com/opengeospatial/ets-wcs11/pull/77

This can be merged now.

**Review by jodygarnett — APPROVED** [review]
This looks good to me, OGC test is now just testing for an exception. The exception message provided by GeoServer is clear enough.

**Pull request merged**
